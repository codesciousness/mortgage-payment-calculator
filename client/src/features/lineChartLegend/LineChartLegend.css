.LineChartLegend {
    width: 20%;
    margin: 1rem;
}