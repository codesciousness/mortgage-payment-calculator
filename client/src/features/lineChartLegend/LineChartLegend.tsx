import React from 'react';
import Box from '@mui/material/Box';
import './LineChartLegend.css';

const styles = {
    borderBottom: 1, 
    borderColor: 'divider'
}

const LineChartLegend = (): JSX.Element => {

    return (
        <section id='LineChartLegend' className='LineChartLegend'>
            <Box sx={styles}>
                <p className='LineChartLegend__label principal'>Principal</p>
            </Box>
            <Box sx={styles}>
                <p className='LineChartLegend__label interest'>Interest</p>
            </Box>
            <Box sx={styles}>
                <p className='LineChartLegend__label balance'>Balance</p>
            </Box>
        </section>
    );
};

export default LineChartLegend;