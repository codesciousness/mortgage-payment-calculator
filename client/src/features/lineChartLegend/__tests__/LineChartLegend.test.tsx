import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import LineChartLegend from '../LineChartLegend';

describe('<LineChartLegend />', () => {
    test('LineChartLegend component is defined', () => {
        expect(<LineChartLegend/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<LineChartLegend/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('it renders correctly', () => {
        render(<LineChartLegend/>);
        expect(screen.getByText('Principal')).toBeInTheDocument();
        expect(screen.getByText('Interest')).toBeInTheDocument();
        expect(screen.getByText('Balance')).toBeInTheDocument();
    });
});