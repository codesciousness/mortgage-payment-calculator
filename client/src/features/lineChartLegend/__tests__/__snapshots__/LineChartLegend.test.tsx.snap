// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineChartLegend /> component matches the snapshot 1`] = `
<section
  className="LineChartLegend"
  id="LineChartLegend"
>
  <div
    className="MuiBox-root css-1gsv261"
  >
    <p
      className="LineChartLegend__label principal"
    >
      Principal
    </p>
  </div>
  <div
    className="MuiBox-root css-1gsv261"
  >
    <p
      className="LineChartLegend__label interest"
    >
      Interest
    </p>
  </div>
  <div
    className="MuiBox-root css-1gsv261"
  >
    <p
      className="LineChartLegend__label balance"
    >
      Balance
    </p>
  </div>
</section>
`;
